File: 0088-merge-sorted-array/0088-merge-sorted-array.js
/**
 * @param {number[]} nums1
 * @param {number} m
 * @param {number[]} nums2
 * @param {number} n
 * @return {void} Do not return anything, modify nums1 in-place instead.
 */
var merge = function(nums1, m, nums2, n) {
    let nums01 = Array.from(nums1);
    let len = nums01.length
     nums01=nums01.slice(0,(m));
    console.log(nums01);
    if (m==0){
        nums1.length=0;
        nums1.push(...nums2);
    }
    else if(n==0){

    }
    else{
        for(i=0;i<n;i++){
                //  console.log(nums2[i]);
            let isAdded=0;
            for(j=0;j<nums01.length;j++){

            if (nums2[i]<=nums01[j]){
                //  console.log(nums1[j]);
                nums01.splice(j,0,nums2[i]);
                isAdded=1;
                break;
            }
            }
            if(!isAdded){
                nums01.push(nums2[i]);
            }
        //  console.log(nums1);
        }
        console.log(nums01);
        nums1.length=0;
        nums1.push(...nums01);
    }
};

File: 0088-merge-sorted-array/0088-merge-sorted-array.py
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        def merge_sort(arr):
            if len(arr)>1:
                nums01=arr[:len(arr)//2]
                nums02=arr[len(arr)//2:]

                # recursion
                merge_sort(nums01)            
                merge_sort(nums02)

                # merge
                i=0
                j=0
                k=0
                
                while i<len(nums01) and j<len(nums02):
                    if nums01[i]<nums02[j]:
                        arr[k]=nums01[i]
                        i+=1
                    elif nums02[j]<=nums01[i]:
                        arr[k]=nums02[j]
                        j+=1
                    k+=1

                while i<len(nums01):
                    arr[k]=nums01[i]
                    i+=1
                    k+=1
                while j<len(nums02):
                    arr[k]=nums02[j]
                    j+=1
                    k+=1
            return arr
        l=merge_sort(nums1[:m]+nums2[:n])
        nums1.clear()
        nums1.extend(l)